mpfilepicker/src/iosMain/kotlin/com/darkrockstudios/libraries/mpfilepicker/IosFilePicker.kt

```diff
@@ -0,0 +1,101 @@
+package com.darkrockstudios.libraries.mpfilepicker
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.remember
+import platform.Foundation.NSURL
+import platform.UIKit.UIApplication
+import platform.UIKit.UIDocumentPickerDelegateProtocol
+import platform.UIKit.UIDocumentPickerViewController
+import platform.UniformTypeIdentifiers.UTType
+import platform.UniformTypeIdentifiers.UTTypeFolder
+import platform.darwin.NSObject
+
+public data class IosFile(
+	override val path: String,
+	override val platformFile: NSURL,
+) : MPFile<NSURL>
+
+private class DocumentPickerDelegate(
+	private val onFileSelected: (NSURL?) -> Unit,
+) : NSObject(),
+	UIDocumentPickerDelegateProtocol {
+	override fun documentPicker(
+		controller: UIDocumentPickerViewController,
+		didPickDocumentsAtURLs: List<*>
+	) {
+		onFileSelected(didPickDocumentsAtURLs.firstOrNull() as? NSURL)
+	}
+
+	override fun documentPickerWasCancelled(controller: UIDocumentPickerViewController) {
+		onFileSelected(null)
+	}
+}
+
+@Composable
+private fun rememberDocumentPickerViewController(
+	onFileSelected: (NSURL?) -> Unit,
+	fileTypes: List<*>,
+	initialDirectory: String?
+): UIDocumentPickerViewController {
+	// Remember the delegate, it gets deallocated if not
+	// stored outside the function, see: https://stackoverflow.com/a/59976378/2673831
+	val delegate = remember { DocumentPickerDelegate(onFileSelected) }
```

Comment:
Is it possible to do all file picking without `remember` calls? it looks like to be compose specific but in future I'd like to split this library and having less compose functions called will make it easy. Here is the issue for it https://github.com/Wavesonics/compose-multiplatform-file-picker/issues/18

Reply:
Will explore options here, there's probably another way to write this without remember calls

Reply:
@shalva97 you guys already have an idea how you want the non-compose API to look like?